### PR TITLE
feat: Report Jest coverage on PRs

### PR DESCRIPTION
**Pull request by thiskevinwang**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

This adds jest coverage reporting + sticky PR commenting.

Note: There is a bug in `jest-coverage-report-action` that ignores `threshold` values, and defaults to `60`, so our reports will always show up as 🔴 
- https://github.com/ArtiomTr/jest-coverage-report-action/blob/main/src/stages/createReport.ts#L37

### CODE REVIEW

**Comment by thiskevinwang** [comment]
It looks like as long as `undefined` is being hardcoded here in the coverage reporting action, `threshold` will always default to `60`

https://github.com/ArtiomTr/jest-coverage-report-action/blob/main/src/stages/createReport.ts#L37

File: jest.config.js [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+/** @type {import('jest').Config} */
+const config = {
 	setupFilesAfterEnv: ['<rootDir>/.test/setup-jest.js'],
 	roots: ['config', 'src', 'build-libs', 'stylelint-rules'],
 	collectCoverageFrom: [
 		'**/*.{js,jsx,ts,tsx}',
 		'!**/*.d.ts',
 		'!**/node_modules/**',
 	],
+	// https://jestjs.io/docs/configuration#coveragethreshold-object
```
Comment (thiskevinwang) [inline jest.config.js:12]:
```suggestion
	// https://jestjs.io/docs/configuration#coveragethreshold-object
	// Note: There is a bug in jest-coverage-report-action that 
	// causes this coverageThreshold to be ignored, and defaulting to 60.
	// - https://github.com/ArtiomTr/jest-coverage-report-action/blob/main/src/stages/createReport.ts#L37
```

**Comment by mhonnatti** [comment]
@BRKalow it looks like @thiskevinwang 's research has landed on the 🔴 being an issue with the way the coverage reporter GH action handles thresholds and our low coverage level. I'm okay okay with this "bug" for the time being. your thoughts?

**Comment by thiskevinwang** [comment]
> @BRKalow it looks like @thiskevinwang 's research has landed on the 🔴 being an issue with the way the coverage reporter GH action handles thresholds and our low coverage level. I'm okay okay with this "bug" for the time being. your thoughts?

@mhonnatti — (@BRKalow highlighted this bug last week)

**Review by mhonnatti — APPROVED** [review]

**Review by BRKalow — CHANGES_REQUESTED** [review]
Thanks for taking a stab at this! I think if we can't figure out how to not make it red every time, we should explore another action or at least wait until we're through with our crunch to ship for GA. Otherwise I think it'll just be extra noise and a distraction that's not actionable.

**Pull request closed without merging**